### ISSUE TITLE
got Floating point exception when change -knl_seq_length_trunc from 200 to 100

### ISSUE THREAD

**Issue by happyMinionVic** [lizekang/ITDD#7]
I had run the code successfully with the given config. But when I tried to reduce the knowledge sentence padding length in  the preprocess by reducing the padding length of knl from 200 to 100,  I got an  `Floating point exception (core dumped)` error and no other traceback. Could you figure out how could this happed and be fixed ? Thanks.

**Comment by happyMinionVic** [comment]
By the way, the error occurs in the start of training rather than running preprocess.py

**Comment by lizekang** [comment]
Could you provide the other configs when you run the preprocess.py?

**Comment by happyMinionVic** [comment]
Here is the config, the only difference is `-knl_seq_length_trunc 100`:
`python preprocess.py \
    --train_src data/src-train-tokenized.txt \
    --valid_src data/src-valid-tokenized.txt \
    --train_knl data/knl-train-tokenized.txt \
    --valid_knl data/knl-valid-tokenized.txt \
    --train_tgt data/tgt-train-tokenized.txt \
    --valid_tgt data/tgt-valid-tokenized.txt \
    --save_data data/cmu_movie \
    -dynamic_dict \
    -share_vocab \
    -src_seq_length_trunc 50 \
    -tgt_seq_length_trunc 50 \
    -knl_seq_length_trunc 100 \
    -src_seq_length 150 \
    -knl_seq_length 800 `

**Comment by lizekang** [comment]
You can try -knl_seq_length 400 and modify the code in onmt/encoders/mtransformer.py. Note that we use three previous dialogs as context so that the knl_seq_length is quadruple of knl_seq_length_trunc.

**Comment by happyMinionVic** [comment]
Ok, I will try this right now. And is there any other restrictions on the sequence length in the preprocessing? Such as `-src_seq_length` is three times as `-src_seq_length_trunc `?

**Comment by lizekang** [comment]
Yes. You can modify the code in `onmt/encoders/mtransformer.py` to change this setting.

**Comment by happyMinionVic** [comment]
The error is still there when I change the `-knl_seq_length` to 400. I run `preprocess.py` with the following config:
`python preprocess.py     --train_src data/src-train-tokenized.txt     --valid_src data/src-valid-tokenized.txt     --train_knl data/knl-train-tokenized.txt     --valid_knl data/knl-valid-tokenized.txt     --train_tgt data/tgt-train-tokenized.txt     --valid_tgt data/tgt-valid-tokenized.txt     --save_data data/cmu_movie     -dynamic_dict     -share_vocab     -src_seq_length_trunc 50     -tgt_seq_length_trunc 50     -knl_seq_length_trunc 100     -src_seq_length 150     -knl_seq_length 400`
Then I initial the training process by running:
`CUDA_VISIBLE_DEVICES=6 python train.py -config config/config-transformer-base-1GPU.yml`
I get the same error:
![image](https://user-images.githubusercontent.com/56298244/66393138-58143980-ea04-11e9-9c88-a38857adeab6.png)

**Comment by happyMinionVic** [comment]
Is the following code should also be modified?
![image](https://user-images.githubusercontent.com/56298244/66393338-d5d84500-ea04-11e9-951b-9ed81041f708.png)

**Comment by lizekang** [comment]
Yes

**Comment by happyMinionVic** [comment]
Where is the knl[600:800 ]? Are they just ignored?

**Comment by lizekang** [comment]
No, knl[600:800] is used in the deocding process.

**Comment by happyMinionVic** [comment]
I had modified all the related parameters in  `mtransformer.py` and still got the the error `Floating point exception`
My pytorch is 1.0.0 and torchtext is 0.4.0.
Where is this bug from ?

**Comment by happyMinionVic** [comment]
Here is the preprocessing config:
`python preprocess.py     --train_src data/src-train-tokenized.txt     --valid_src data/src-valid-tokenized.txt     --train_knl data/knl-train-tokenized.txt     --valid_knl data/knl-valid-tokenized.txt     --train_tgt data/tgt-train-tokenized.txt     --valid_tgt data/tgt-valid-tokenized.txt     --save_data data/cmu_movie     -dynamic_dict     -share_vocab     -src_seq_length_trunc 15     -tgt_seq_length_trunc 15     -knl_seq_length_trunc 60     -src_seq_length 45     -knl_seq_length 240`

and the `mtransformer.py`:
![image](https://user-images.githubusercontent.com/56298244/66446871-d2819f80-ea7e-11e9-883f-536f95b00538.png)

**Comment by lizekang** [comment]
Oh, sorry, you should also modify the parameters in `onmt/models/model.py`.

**Comment by happyMinionVic** [comment]
OK, everything goes right. In conclusion，one should modify the number in both `onmt/models/model.py` and `onmt/encoder/mtransformer.py` when change the config during preprocessing the data.
